### ISSUE TITLE
Bug? test_target_data_if -- validation criteria maybe incorrect

### ISSUE THREAD

**Issue by naromero77intel** [SOLLVE/sollve_vv#514]
**Describe the bug**
I believe there is a bug in 4.5/target_data/test_target_data_if.F90

Here is the criteria for the first part of the test ( `tests_target_data_if_simple()` ):
https://github.com/SOLLVE/sollve_vv/blob/master/tests/4.5/target_data/test_target_data_if.F90#L88-L89

but it looks like it should be:
```
+                 IF ( s > THRESHOLD .AND. isOffloading) THEN
```

which would make it symmetrical to the second part of the test (`tests_target_data_if_nested()` ):
https://github.com/SOLLVE/sollve_vv/blob/master/tests/4.5/target_data/test_target_data_if.F90#L168

In our own internal testing, the test passes when the target is the Gen9,
`-fiopenmp -fopenmp-targets=spir64`

but fails if the target is the host:
`-fiopenmp -fopenmp-targets=x86_64`

**Test that it applies to**
4.5/target_data/test_target_data_if.F90

**To Reproduce**
Gen9 test
`ifx -o test_target_data_if.x -c -fiopenmp -fopenmp-targets=spir64 test_target_data_if.F90`
works

Host test
`ifx -o test_target_data_if.x -c -fiopenmp -fopenmp-targets=x86_64 test_target_data_if.F90`
fails

**Expected behavior**
I would have expected this test to pass if the target is the host --- but was it actually designed to work in this manner?

**Compiler**
Intel Fortran Compiler

**Accelerator hardware**
Intel Gen9 GPU

**Comment by tmh97** [comment]
Good catch @naromero77intel and thanks for letting us know! We **do** want this test to pass if target is the host. I am going to re-work this test such that it will pass when offloading is enabled.

I will link this issue and @ you when I push a PR to patch this. Thanks again.

**Comment by naromero77intel** [comment]
Thanks @tmh97 -- I unfortunately was not able to fix the test with the simple change. I think a second edit is needed to get it to run when the target is host.
